server/src/domain/ldap/group.rs

```diff
@@ -126,6 +127,19 @@ fn convert_group_filter(
                         vec![],
                     )))),
                 },
+                "dn" => Ok(
+                    match get_group_id_from_distinguished_name(
```

Comment:
We _could_ write it as
```rust
Ok(
  get_group_id(...)
  .map(GroupRequestFilter::DisplayName)
  .unwrap_or_else(|_| {
    warn!(...);
    GroupRequestFilter::Not(...)
  }))
```
And it might even make sense to add a `false()` and `true()` static functions to `GroupRequestFilter`, but that's for a next PR. Let's keep it as is, I'll make a second pass later.